contractcourt/channel_arbitrator.go

```diff
@@ -244,15 +262,56 @@ func (c *ChannelArbitrator) Start() error {
 		return err
 	}
 
+	// If the channel has been marked pending close in the database, and we
+	// haven't transitioned the state machine to StateContractClosed (or a
+	// suceeding state), then a state transition most likely failed. We'll
+	// try to recover from this by manually advancing the state by setting
+	// the corresponding close trigger.
+	trigger := chainTrigger
+	triggerHeight := uint32(bestHeight)
+	if c.cfg.IsPendingClose {
+		switch c.state {
+		case StateDefault:
+			fallthrough
+		case StateBroadcastCommit:
+			fallthrough
+		case StateCommitmentBroadcasted:
+			switch c.cfg.CloseType {
+			case channeldb.LocalForceClose:
+				trigger = localCloseTrigger
+			case channeldb.RemoteForceClose:
+				trigger = remoteCloseTrigger
+			}
```

Comment:
i think we also want a case for coop closes here, so that we can handle:
1) channel closed by channel arb
2) lnd crash
3) channel still in Default, but has been closed
4) want to end up in FullyResolved

Reply:
Since a coop close is always marked with `CloseChannelSummary{ IsPending: false }` this won't ever be hit. 

However, do you think we should change it to `IsPending: true` for consistency, and make it be fully closed (`IsPending: false`) when `MarkChannelResolved` is called? I originally didn't make this change to be conservative and change behaviour as little as possible.

Reply:
Ended up setting `IsPending: true` also for coop closes.